### PR TITLE
Created an endpoint to get user details

### PR DESCRIPTION
**Pull request by arionrefat**
Created an endpoint GET /v1/user/{userId | username}

### CODE REVIEW

File: packages/server/src/lib/User.ts [hunk 1]
```diff
@@ -75,9 +77,46 @@ async function loginUser(
   return resUser;
 }
 
-export async function getLoggedInUser(userJwtPayload: UserJWTPayload | null | undefined): Promise<UserMeResponse> {
+async function getUserByUsernameOrId(
+  payload: UserGetPayload
```
Comment (xImouto) [inline packages/server/src/lib/User.ts:81]: as the name of the function suggests it should take in an argument of Username or Id
```suggestion
  usernameOrId: string
```

**Review by xImouto — CHANGES_REQUESTED** [review]

File: packages/server/src/controllers/userController.ts [hunk 2]
```diff
@@ -50,13 +57,22 @@ const userController = {
     }
   },
 
+  getByUsernameOrId: async (
+    req: Request<UserGetPayload>,
+    res: Response<ApiResponse<UserGetResponse>>
+  ) => {
+    try {
+      const user = await getUserByUsernameOrId(req.params);
```
Comment (Devorein) [inline packages/server/src/controllers/userController.ts:65]: Change this to `req.params.usernameOrId` as @xImouto suggested

File: packages/server/src/lib/User.ts [hunk 3]
```diff
@@ -75,9 +77,46 @@ async function loginUser(
...
+): Promise<UserGetResponse> {
+  const { usernameOrId } = payload;
+  const isID = checkUUID(usernameOrId);
+  let user; // With my heave heart I have failed you, nothing works :(
```
Comment (Devorein) [inline packages/server/src/lib/User.ts:85]: Use `let user: PublicUser;` 😕

File: packages/server/src/utils/checkUUID.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export default function checkUUID(identifier: string): boolean {
```
Comment (Devorein) [inline packages/server/src/utils/checkUUID.ts:1]: Best to add some comments for the function. Look at some examples in the utils folder

**Review by Devorein — CHANGES_REQUESTED** [review]

File: packages/server/src/lib/User.ts (hunk 3 above)
Comment (arionrefat) [reply packages/server/src/lib/User.ts:85]: I did that actually, it gave an error, Now that I have read it now, i feel dissapointed

File: packages/server/src/lib/User.ts (hunk 1 above)
Comment (arionrefat) [reply packages/server/src/lib/User.ts:81]: image

File: packages/server/src/controllers/userController.ts (hunk 2 above)
Comment (arionrefat) [reply packages/server/src/controllers/userController.ts:65]: Done

File: packages/server/src/lib/User.ts (hunk 1 above)
Comment (arionrefat) [reply packages/server/src/lib/User.ts:81]: Done

File: packages/server/src/utils/checkUUID.ts (hunk 4 above)
Comment (arionrefat) [reply packages/server/src/utils/checkUUID.ts:1]: done

File: packages/server/src/lib/User.ts (hunk 3 above)
Comment (arionrefat) [reply packages/server/src/lib/User.ts]: fixed

**Review by Devorein — APPROVED** [review]

**Review by xImouto — APPROVED** [review]

File: packages/server/src/controllers/userController.ts [hunk 5]
```diff
@@ -50,6 +57,18 @@ const userController = {
...
 
+  getByUsernameOrId: async (
+    req: Request<UserGetPayload>,
+    res: Response<ApiResponse<UserGetResponse>>
+  ) => {
+    try {
+      const user = await getUserByUsernameOrId(req.params.usernameOrId);
+      return handleSuccess<UserGetResponse>(res, user);
+    } catch (err) {
+      handleError(res, err);
```
[the remark this answers is not in the document]
Comment (arionrefat) [reply packages/server/src/controllers/userController.ts]: done

**Pull request merged**